From 24692abf3b03b2d1990a147dae8b6fd7da16c9be Mon Sep 17 00:00:00 2001
From: Yuan Juntao <juntao.yuan@intel.com>
Date: Tue, 17 Mar 2015 11:36:45 +0800
Subject: [PATCH] mmc: host: enable full power cycle feature for sd host
 controller.

Observed a KINGMAX 8GB SDHC memory card has initialization issue.
Current intialization process works well with most sd memory cards,
but the card mentioned above is an exception, sdio_reset operation
confuses it and make it impossible to finish card inner intialization.

So, turn on the FULL POWER CYCLE feature for SD host controller to do
a power cycle before initialization process starts.

Change-Id: If2037e82dd7c24ee1b2e56113f93b66cf90e1206
Signed-off-by: Yuan Juntao <juntao.yuan@intel.com>
---
 drivers/mmc/host/sdhci-acpi.c | 1 +
 1 file changed, 1 insertion(+)

Index: linux/drivers/mmc/host/sdhci-acpi.c
===================================================================
--- linux.orig/drivers/mmc/host/sdhci-acpi.c	2015-03-18 13:17:33.634233754 -0700
+++ linux/drivers/mmc/host/sdhci-acpi.c	2015-03-18 13:19:11.131197894 -0700
@@ -212,6 +212,7 @@
 };
 
 static const struct sdhci_acpi_slot sdhci_acpi_slot_int_sd = {
+	.caps2   = MMC_CAP2_FULL_PWR_CYCLE,
 	.flags   = SDHCI_ACPI_SD_CD | SDHCI_ACPI_RUNTIME_PM,
 	.quirks2 = SDHCI_QUIRK2_PRESET_VALUE_BROKEN,
 	.probe_slot = sdhci_acpi_sd_probe_slot,
